File: messagebus/src/main/java/com/yahoo/messagebus/network/rpc/test/SlobrokState.java
// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.messagebus.network.rpc.test;

import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Set;

/**
 * @author Simon Thoresen Hult
 */
public class SlobrokState {
    private Map<String, Integer> data = new LinkedHashMap<String, Integer>();

    public SlobrokState add(String pattern, int count) {
        data.put(pattern, count);
        return this;
    }

    public Set<String> getPatterns() {
        return data.keySet();
    }

    public int getCount(String pattern) {
        return data.containsKey(pattern) ? data.get(pattern) : 1;
    }
}

File: messagebus/src/vespa/messagebus/testlib/slobrokstate.cpp
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include "slobrokstate.h"

namespace mbus {

SlobrokState::SlobrokState()
    : _data()
{ }

SlobrokState &
SlobrokState::add(const string &pattern, uint32_t cnt)
{
    _data.push_back(std::make_pair(pattern, cnt));
    return *this;
}

SlobrokState::ITR
SlobrokState::begin() const
{
    return _data.begin();
}

SlobrokState::ITR
SlobrokState::end() const
{
    return _data.end();
}

} // namespace mbus
